ecs-cli/modules/utils/compose/ecs_params_reader.go

```diff
@@ -36,15 +38,21 @@ type EcsTaskDef struct {
 	TaskRoleArn          string        `yaml:"task_role_arn"`
 	ContainerDefinitions ContainerDefs `yaml:"services"`
 	ExecutionRole        string        `yaml:"task_execution_role"`
-	TaskSize             TaskSize      `yaml:"task_size"`           // Needed to run FARGATE tasks
+	TaskSize             TaskSize      `yaml:"task_size"` // Needed to run FARGATE tasks
 }
 
+// ContainerDefs is a map of ContainerDefs within a task definition
 type ContainerDefs map[string]ContainerDef
 
+// ContainerDef specifies settings for a specific container
 type ContainerDef struct {
-	Essential bool `yaml:"essential"`
+	Essential         bool  `yaml:"essential"`
+	CPUShares         int64 `yaml:"cpu_shares"`
+	Memory            int64 `yaml:"memory"`
+	MemoryReservation int64 `yaml:"memory_reservation"`
 }
 
+// TaskSize specifies task-level resource limits
```

Comment:
Ha, I had literally added a similar comment in the PR i'm working on but looks like you beat me to it :)  I'd maybe specify: "Task Size holds Cpu and Memory values needed for Fargate tasks", since these values have to correspond to the supported buckets for each (cf: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-cpu-memory-error.html)